torchao/quantization/qat/embedding.py

```diff
@@ -85,6 +82,27 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
             self.sparse,
         )
 
+    @classmethod
+    def from_embedding(
+        cls,
+        mod: torch.nn.Embedding,
+        weight_config: Optional[FakeQuantizeConfig] = None,
+    ):
+        new_embedding = FakeQuantizedEmbedding(
+            mod.num_embeddings,
+            mod.embedding_dim,
+            mod.padding_idx,
+            mod.max_norm,
+            mod.norm_type,
+            mod.scale_grad_by_freq,
+            mod.sparse,
+            weight_config=weight_config,
+            device=mod.weight.device,
+        )
+        if mod.weight.device != torch.device("meta"):
```

Comment:
qq: why is this check needed?

Reply:
added a comment, basically this would error in the distributed setting since the model is instantiated on the meta device for non-zero rank